### ISSUE TITLE
Solaris 11 fails for derived-key-test

### ISSUE THREAD

**Issue by quanah** [heimdal/heimdal#315]
```
derive_key failed (enctype 16)
should be: 925179d04591a79b5d3192c4a7e9c289b049c71f6ee604cd
result was: 925179d04591a79b5d15f70138df0185baef15f70138df0e
derive_key failed (enctype 16)
should be: 9e58e5a146d9942a101c469845d67a20e3c4259ed913f207
result was: 9e58e5a146d9942a10ce135d5b1f3b898f91ce135d5b1f15
derive_key failed (enctype 16)
should be: 13fef80d763e94ec6d13fd2ca1d085070249dad39808eabf
result was: 13fef80d763e94ec6da4674f08d5f1fe1ff7a4674f08d5fd
derive_key failed (enctype 16)
should be: f8dfbf04b097e6d9dc0702686bcb3489d91fd9a4516b703e
result was: f8dfbf04b097e6d9dc682089ef0e31d0ef3d682089ef0ea4
derive_key failed (enctype 16)
should be: 2370da575d2a3da864cebfdc5204d56df779a7df43d9da43
result was: 2370da575d2a3da8648af461015dada1927c8af461015d40
derive_key failed (enctype 16)
should be: 0126388aadc81a1f2a62bc45f8d5c19151bacdd5cb798a3e
result was: 0126388aadc81a1f2ad0e90bc8984fbc1361d0e90bc8987a
derive_key failed (enctype 16)
should be: 348057ec98fdc48016161c2a4c7a943e92ae492c989175f7
result was: 348057ec98fdc480169b3492bc1ccb8f2c159b3492bc1c1f
derive_key failed (enctype 16)
should be: a8808ac267dada3dcbe9a7c84626fbc761c294b01315e5c1
result was: a8808ac267dada3dcb68524026cd575b629b68524026cdb5
derive_key failed (enctype 16)
should be: c813f88a3be3b334f75425ce9175fbe3c8493b89c8703b49
result was: c813f88a3be3b334f7941f0d5bb5704a7357941f0d5bb594
derive_key failed (enctype 16)
should be: f48ffd6e83f83e7354e694fd252cf83bfe58f7d5ba37ec5d
result was: f48ffd6e83f83e7354341531b63d6b6b8968341531b63dd0
FAIL: derived-key-test
```

**Comment by sfhacker** [comment]
Just built master branch from source and got this:

PASS derived-key-test.exe (exit status: 0)

**Comment by dariaphoebe** [comment]
Works when using hcrypto instead of pkcs11 as crypto provider in evp.c

**Comment by quanah** [comment]
@lhoward This also appears to be fixed by 2b5b7cf

**Comment by lhoward** [comment]
Fixed in a88d0066.
